### PR TITLE
Add github api to support setting status by checking pull requests

### PR DESCRIPTION
**Pull request by taquitos**
Added api to github_service:
- open_pull_requests
- last_commit_sha_for_all_open_pull_requests
- status_for_commit_sha
- update_all_open_prs_without_status_to_pending_status!

### CODE REVIEW

File: services/code_hosting/git_hub_service.rb [hunk 1]
```diff
@@ -79,7 +104,8 @@ def set_build_status!(repo: nil, sha: nil, state: nil, target_url: nil, descript
       # https://octokit.github.io/octokit.rb/Octokit/Client/Statuses.html
 
       task = TaskQueue::Task.new(work_block: proc {
-        logger.debug("Setting status #{state} on #{target_url}")
+        state_details = target_url.nil? ? "#{repo}, sha #{sha}" : target_url
```
Comment (taquitos) [inline services/code_hosting/git_hub_service.rb]: Clearer logging

File: services/code_hosting/git_hub_service.rb [hunk 2]
```diff
@@ -36,6 +36,36 @@ def username
...
+      return client.statuses(repo_full_name, sha)
+    end
+
+    # updates the most current commit to "pending" on all open prs if they don't have a status.    
+    # returns a list of commits that have been updated to `pending` status
+    def update_all_open_prs_without_status_to_pending_status!(repo_full_name: nil)
+      open_pr_commits = self.last_commit_sha_for_all_open_pull_requests(repo_full_name: repo_full_name)
+      updated_commits = []
+      open_pr_commits.each do |sha|
+        statuses = self.status_for_commit_sha(repo_full_name: repo_full_name, sha: sha)
```
Comment (KrauseFx) [inline services/code_hosting/git_hub_service.rb]: Does this fetch all statuses, or just the statuses created by `fastlane.ci`?

Comment (taquitos) [reply services/code_hosting/git_hub_service.rb]: alllll

Comment (KrauseFx) [reply services/code_hosting/git_hub_service.rb]: Let me append a question here: should we automatically remove all non-fastlane-ci build status in the `status_for_commit_sha` method, to make things work?

Comment (taquitos) [reply services/code_hosting/git_hub_service.rb]: I have a TODO: in some code I'm working on right now.
Yes, we should filter out statuses not created by fastlane.ci 🚫

Comment (KrauseFx) [reply services/code_hosting/git_hub_service.rb]: perfect, thanks 👍

**Pull request merged**